utils/writeable/src/lib.rs

```diff
@@ -137,12 +138,47 @@ pub trait Writeable {
     /// Creates a new String with the data from this Writeable. Like ToString,
     /// but smaller and faster.
     ///
-    /// Not intended to be overriden.
-    fn writeable_to_string(&self) -> String {
+    /// The default impl allocates an owned String. However, if it is possible to return a
+    /// borrowed string, overwrite this method to return a `Cow::Borrowed`.
+    ///
+    /// To remove the `Cow` wrapper, call `.into_owned()`.
+    ///
+    /// # Examples
+    ///
+    /// Inspect a `Writeable` before writing it to the sink:
+    ///
+    /// ```
+    /// use writeable::Writeable;
+    /// use core::fmt::{Write, Result};
+    ///
+    /// fn write_if_ascii<W, S>(w: &W, sink: &mut S) -> Result
+    /// where
+    ///     W: Writeable + ?Sized,
+    ///     S: Write + ?Sized,
+    /// {
+    ///     let s = w.writeable_to_string();
+    ///     if s.is_ascii() {
+    ///         sink.write_str(&s)
+    ///     } else {
+    ///         Ok(())
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// Convert the `Writeable` into a fully owned `String`:
+    ///
+    /// ```
+    /// use writeable::Writeable;
+    ///
+    /// fn make_string(w: &impl Writeable) -> String {
+    ///     w.writeable_to_string().into_owned()
+    /// }
+    /// ```
+    fn writeable_to_string(&self) -> Cow<str> {
```

Comment:
Maybe `writeable_to_str` is a better name to highlight it doesn't always allocate?

I assume we currently have the `writeable_` prefix to avoid confusion with `Display`, so with `str` we could drop that prefix.

Reply:
I filed #1453 to discuss names.  I did not want to change names in this PR because it touches a lot more files and I wanted to have the name discussion after making the signature change.